snuba/query/processors/granularity_processor.py

```diff
@@ -30,8 +35,32 @@ def __get_granularity(query: Query) -> int:
             f"Granularity must be multiple of one of {GRANULARITIES_AVAILABLE}"
         )
 
+    @staticmethod
+    def __get_granularity_for_subscription(query: Query) -> Optional[int]:
+        granularity = None
+        entity = get_entity(query.get_from_clause().key)
+        if not entity.required_time_column:
+            raise InvalidGranularityException(
+                "Entity mush have a time column defined to be able to calculate granularity"
+            )
+        from_date, to_date = get_time_range(query, entity.required_time_column)
+        if from_date and to_date:
+            time_interval = to_date - from_date
+            if time_interval <= timedelta(minutes=60):
+                granularity = 10
+            elif time_interval <= timedelta(hours=4):
+                granularity = 60
+            else:
+                granularity = 60 * 60
+        return granularity
+
     def process_query(self, query: Query, request_settings: RequestSettings) -> None:
-        granularity = self.__get_granularity(query)
+        granularity = None
+        if isinstance(request_settings, SubscriptionRequestSettings):
+            granularity = self.__get_granularity_for_subscription(query)
```

Comment:
Should we raise an exception in this branch if `query.get_granularity()` is not None? It might be surprising to get a different granularity than expected.

Reply:
For subscriptions? there is no way for a user to define a granularity on a crash rate alert anyway